src/H5FDsubfiling/H5FDsubfiling.c

```diff
@@ -354,13 +354,7 @@ H5FD_subfiling_init(void)
                     "Subfiling VFD requires the use of MPI_Init_thread with MPI_THREAD_MULTIPLE");
         }
         else {
-            char *env_var;
-            int   required = MPI_THREAD_MULTIPLE;
-
-            /* Ensure that Subfiling VFD has been loaded dynamically */
-            env_var = HDgetenv(HDF5_DRIVER);
-            if (!env_var || HDstrcmp(env_var, H5FD_SUBFILING_NAME))
-                H5_SUBFILING_GOTO_ERROR(H5E_VFL, H5E_CANTINIT, H5I_INVALID_HID, "MPI isn't initialized");
```

Comment:
If the Subfiling VFD is called from serial code (the tools code in this case), it always needs to initialize MPI, even when it wasn't dynamically loaded.